src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -735,15 +735,19 @@ private void checkParam(int n) throws SQLServerException {
     @Override
     public String getParameterClassName(int param) throws SQLServerException {
         checkClosed();
+        String parameterClassName = null;
         try {
             if (procMetadata == null) {
                 // PreparedStatement.
                 checkParam(param);
-                return queryMetaMap.get(param).parameterClassName;
+                parameterClassName = queryMetaMap.get(param).parameterClassName;
             } else {
-                JDBCType jdbcType = JDBCType.of((short) getParameterInfo(param).get("DATA_TYPE"));
-                return jdbcType.className();
+                Map<String, Object> info = getParameterInfo(param);
+                if (null != info) {
+                    parameterClassName = JDBCType.of((short) info.get("DATA_TYPE")).className();
+                }
```

Comment:
So if getParameterInfo(param) returns null, we won't call any APIs which may cause a nullptr exception. However, the code in its current state will return a String object that's `null` in these scenarios instead of an actual empty String such as `""`. Do we want this behavior? If there are calls to `String.length()` using the String returned by these methods, we'll get nullptr issues again.

Reply:
i don't think that's the point of this change. the objective of this change is to avoid calling getParameterInfo(param).get("DATA_TYPE"), which assumes that getParameterInfo(param) is not going to be null (which it could). The PR is not concerned with how the String returned from this method is consumed, it's the caller's responsibility to check for nulls from this method (which is could return before, anyways).

Reply:
I don't think this change would be out of scope for the PR. Worst case scenario we could just  backlog it, but I don't think the fix is complicated enough for that. Returning `null` from this method makes no sense. We should either be throwing an error or returning `""`. According to javadocs, the function `Retrieves the fully-qualified name of the Java class whose instances should be passed to the method PreparedStatement.setObject`. `null` is not a valid "fully-qualified name" and the value is expected to be used with `setObject`, which may result in confusion as null is a "valid" value for any Object class and not only String.

Reply:
you're probably overthinking it, the method was already returning null before. I imagine this PR was made to account for the static analysis tool results, and I think the PR is aimed at solving that complaint. Not to mention that changing the method to return an empty string instead of null is a behavioral change, and it could potentially break existing applications.

Reply:
It's not really about overthinking, it's just bad design. We'll probably have to take a call on the backwards compatibility thing, but returning `null` when there's no reason to isn't good practice. Creating PRs to please static code analysis tools but ignoring bad practice is a little convoluted since that's why we even use these tools.

Reply:
I think these APIs are returning null since beginning by design.. should rather throw exception. Will fix this behavior overall.

Reply:
Returning an empty string instead of a null when we do not have a corresponding entry for the class name isn't necessarily a better design. If the required information hasn't been provided, we should return it as is (null), instead of returning an instance of something that doesn't exist and giving it an empty String. Again, this is a public API that's being used by customers, and it could break existing applications for not a very strong reason.